packages/nx/src/daemon/client/client.ts

```diff
@@ -295,7 +295,7 @@ export class DaemonClient {
       this.currentResolve = resolve;
       this.currentReject = reject;
 
-      this.socketMessenger.sendMessage(message);
+      this.socketMessenger?.sendMessage(message);
```

Comment:
The `this._connected = true` is in the wrong place.

It should come after the server has started in the background.

Then it will happen synchronously with this call and `this.socketMessenger` will always be defined for this.